src/github/createPRViewProvider.ts

```diff
@@ -0,0 +1,258 @@
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { FolderRepositoryManager, titleAndBodyFrom } from './folderRepositoryManager';
+import webviewContent from '../../media/createPR-webviewIndex.js';
+import { getNonce, IRequestMessage, WebviewBase } from '../common/webview';
+import { PullRequestDescriptionSource, PullRequestDescriptionSourceEnum, PullRequestTitleSource, PullRequestTitleSourceEnum } from '../view/quickpick';
+import * as PersistentState from '../common/persistentState';
+import { PR_SETTINGS_NAMESPACE, PR_TITLE } from '../common/settingKeys';
+import { OctokitCommon } from './common';
+import { PullRequestModel } from './pullRequestModel';
+
+interface RemoteInfo {
+	owner: string;
+	repositoryName: string;
+}
+
+export class CreatePullRequestViewProvider extends WebviewBase implements vscode.WebviewViewProvider {
+	public static readonly viewType = 'github:createPullRequest';
+
+	private _webviewView: vscode.WebviewView;
+
+	private _onDone = new vscode.EventEmitter<PullRequestModel | undefined> ();
+	readonly onDone: vscode.Event<PullRequestModel | undefined> = this._onDone.event;
+
+	constructor(
+		private readonly _extensionUri: vscode.Uri,
+		private readonly _folderRepositoryManager: FolderRepositoryManager,
+		private readonly _isDraft: boolean
+	) {
+		super();
+	}
+
+	public resolveWebviewView(
+		webviewView: vscode.WebviewView,
+		context: vscode.WebviewViewResolveContext,
+		_token: vscode.CancellationToken,
+	) {
+
+		this._webviewView = webviewView;
+		this._webview = webviewView.webview;
+		super.initialize();
+		webviewView.webview.options = {
+			// Allow scripts in the webview
+			enableScripts: true,
+
+			localResourceRoots: [
+				this._extensionUri
+			]
+		};
+
+		webviewView.webview.html = this._getHtmlForWebview();
+
+		this.initializeParams();
+	}
+
+	public show() {
+		this._webviewView.show();
+	}
+
+	private async getTitle(): Promise<string> {
+		const method = vscode.workspace.getConfiguration(PR_SETTINGS_NAMESPACE).get<PullRequestTitleSource>(PR_TITLE, PullRequestTitleSourceEnum.Ask);
+
+		switch (method) {
+
+			case PullRequestTitleSourceEnum.Branch:
+				return this._folderRepositoryManager.repository.state.HEAD!.name!;
+
+			case PullRequestTitleSourceEnum.Commit:
+				return titleAndBodyFrom(await this._folderRepositoryManager.getHeadCommitMessage()).title;
+
+			default:
+				return '';
+		}
+	}
+
+	private async getDescription(): Promise<string> {
+		const method = vscode.workspace.getConfiguration('githubPullRequests').get<PullRequestDescriptionSource>('pullRequestDescription', PullRequestDescriptionSourceEnum.Ask);
+
+		switch (method) {
+
+			case PullRequestDescriptionSourceEnum.Template:
+				const templateUris = await this._folderRepositoryManager.getPullRequestTemplates();
+				if (templateUris[0]) {
```

Comment:
I've been doing a length check before accessing `[0]`. TIL that this was an option! 😅